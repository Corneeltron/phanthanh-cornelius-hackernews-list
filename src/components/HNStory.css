* {
  font-size: 14px;
}

a {
  padding-left: 1em;
}

li {
  color: #5f5f5f;
  font-weight: 500;
  margin-bottom: 1em;
}

.links {
  color: black;
  text-decoration: none;
}

.links:visited {
  color: red;
}

a:hover {
  color: blue;
}

.link-info-container {
  color: #5f5f5f;
  margin-bottom: .5em;
  padding-left: 1em;
}

.link-info {
  font-size: .75em;
  text-decoration: none;
  color: #5f5f5f;
  padding-left: 0;
}
