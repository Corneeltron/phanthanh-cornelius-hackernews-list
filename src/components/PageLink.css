.page-link {
  flex: inline-flex;
  border: 1px solid #dee2e6;
  background: #fff;
  padding: 10px 15px;
  color: #0d6efd;
  font-size: 16px;
  font-weight: 500;
  text-decoration: none;
  cursor: pointer;
}

.page-link:first-child {
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.page-link:last-child {
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.page-link:not(:first-child) {
  margin-left: -1px;
}

.page-link:hover,
.page-link:focus {
  color: #0a58ca;
  background-color: #e9ecef;
}

.page-link.active {
  color: #fff;
  border-color: #0d6efd;
  background-color: #0d6efd;
}