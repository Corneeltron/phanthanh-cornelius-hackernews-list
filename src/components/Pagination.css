.pagination {
    display: flex;
    padding-bottom: 1em;
}