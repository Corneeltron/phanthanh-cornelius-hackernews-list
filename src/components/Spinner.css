:root {
  --sk-size: 30px;
  --sk-color: #ff6600;
}

.spinner-container {
  margin-top: 2em;
  margin-bottom: 2em;
}

.spinner {
  margin: auto;
  width: var(--sk-size);
  height: var(--sk-size);
  position: relative; 
}

.spinner-dot {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0; 
}

.spinner-dot:before {
  content: '';
  display: block;
  width: 15%;
  height: 15%;
  background-color: var(--sk-color);
  border-radius: 100%;
  animation: spinner 1.2s infinite ease-in-out both; 
}

.spinner-dot:nth-child(1)  { transform: rotate(30deg);  }
.spinner-dot:nth-child(2)  { transform: rotate(60deg);  }
.spinner-dot:nth-child(3)  { transform: rotate(90deg);  }
.spinner-dot:nth-child(4)  { transform: rotate(120deg); }
.spinner-dot:nth-child(5)  { transform: rotate(150deg); }
.spinner-dot:nth-child(6)  { transform: rotate(180deg); }
.spinner-dot:nth-child(7)  { transform: rotate(210deg); }
.spinner-dot:nth-child(8)  { transform: rotate(240deg); }
.spinner-dot:nth-child(9)  { transform: rotate(270deg); }
.spinner-dot:nth-child(10) { transform: rotate(300deg); }
.spinner-dot:nth-child(11) { transform: rotate(330deg); }
.spinner-dot:nth-child(1):before  { animation-delay: -1.1s; }
.spinner-dot:nth-child(2):before  { animation-delay: -1.0s; }
.spinner-dot:nth-child(3):before  { animation-delay: -0.9s; }
.spinner-dot:nth-child(4):before  { animation-delay: -0.8s; }
.spinner-dot:nth-child(5):before  { animation-delay: -0.7s; }
.spinner-dot:nth-child(6):before  { animation-delay: -0.6s; }
.spinner-dot:nth-child(7):before  { animation-delay: -0.5s; }
.spinner-dot:nth-child(8):before  { animation-delay: -0.4s; }
.spinner-dot:nth-child(9):before  { animation-delay: -0.3s; }
.spinner-dot:nth-child(10):before { animation-delay: -0.2s; }
.spinner-dot:nth-child(11):before { animation-delay: -0.1s; }

@keyframes spinner {
  0%, 39%, 100% {
    opacity: 0;
    transform: scale(0.6);
  } 40% {
    opacity: 1; 
    transform: scale(1);
  }
}