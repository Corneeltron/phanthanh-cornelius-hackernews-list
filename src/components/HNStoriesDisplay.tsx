import "./HNStoriesDisplay.css";
import { HNStory } from "../api/HNStory";

interface Props {
  stories: HNStory[];
}

export const HNStoriesDisplay = (props: Props ) => {
  
  const { stories } = props;

  if (stories.length === 0) {
    return <div>Loading...</div>;
  }

  return (
    <div>
      <ul>
        {stories.map((post) => (
          <li key={post.id}>
            <a href={post.url}>{post.title}</a>
          </li>
        ))}
      </ul>
    </div>
  );
};
