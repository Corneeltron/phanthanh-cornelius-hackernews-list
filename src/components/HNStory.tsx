import { useState, useEffect } from 'react';
import './HNStory.css';

export interface HNStory {
  by: string;
  id: number;
  descendants: number;
  kid: number[];
  score: number;
  time: number;
  title: string;
  type: string;
  url: string;
}

type HNStoryPageProps = {
  id: number;
  currentPage: number;
};

export const HNStoryPage = ({ id, currentPage }: HNStoryPageProps) => {
  const [storyByIdData, setStoryByIdData] = useState<HNStory>({
    by: "default",
    id: 0,
    descendants: 0,
    kid: [],
    score: 0,
    time: 0,
    title: "title",
    type: "story",
    url: "www.ycombinator.com",
  });
  const [error, setError] = useState<string | boolean>(false)

  useEffect(() => {
    async function getStoriesById(id: number) {
      try {
        fetch(`https://hacker-news.firebaseio.com/v0/item/${id}.json`)
          .then((res) => res.json())
          .then((data) => setStoryByIdData(data));

      } catch (err) {
        if (err instanceof Error) {
          setError(err.message);
        } else {
          setError(JSON.stringify(err))
        }
      }
    }
    getStoriesById(id);
  }, [currentPage, id]); 

  return (
    <div key={storyByIdData.id}>
      <a target="_blank" rel="noopener noreferrer" className="links" href={storyByIdData.url}>
        {storyByIdData.title}
      </a>
      <div className="link-info-container">
        <span className="link-info">{storyByIdData.score} points by </span>
        <a
          target="_blank"
          href={`https://news.ycombinator.com/user?id=${storyByIdData.by}`}
          rel="noopener noreferrer"
          className="link-info"
        >
          {`${storyByIdData.by} `}
        </a>
        <span className="link-info">{new Date(storyByIdData.time * 1000).getHours()} hours ago </span>
        |
        <a
          target="_blank"
          href={`https://news.ycombinator.com/item?id=${storyByIdData.id}`}
          rel="noopener noreferrer"
          className="link-info"
        >
          {" "}
          {`${storyByIdData.descendants} comments`}
        </a>
      </div>
      {error && <p>An error has occurred loading this story!</p>}
    </div>
  );
};
