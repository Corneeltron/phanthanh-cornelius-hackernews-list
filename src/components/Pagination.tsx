import { getPaginationItems } from "../lib/pagination";
import { PageLink } from "./PageLink";
import './Pagination.css';

export type Props = {
  currentPage: number;
  lastPage: number;
  maxLength: number;
  onPageChange: (page: number) => void;
};

export const Pagination = ({
  currentPage,
  lastPage,
  maxLength,
  onPageChange,
}: Props) => {
  const pageNums = getPaginationItems(currentPage, lastPage, maxLength);
  return (
    
    <nav className="pagination" aria-label="Pagination">
      <PageLink
        href="#"
        disabled={currentPage === 1}
        onClick={() => onPageChange(currentPage - 1)}
      >
        Previous
      </PageLink>
      {pageNums.map((pageNum, idx) => (
        <PageLink
          key={idx}
          href="#"
          active={pageNum === currentPage}
          disabled={isNaN(pageNum)}
          onClick={() => onPageChange(pageNum)}
        >
          {!isNaN(pageNum) ? pageNum : '...'}
        </PageLink>
      ))}
      <PageLink
        href="#"
        disabled={currentPage === lastPage}
        onClick={() => onPageChange(currentPage + 1)}
      >
        Next
      </PageLink>
    </nav>
  );
};
