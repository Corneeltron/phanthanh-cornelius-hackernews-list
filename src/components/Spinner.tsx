import './Spinner.css';

export const Spinner = () => (
  <div className="spinner-container">
    <div className="spinner">
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
      <div className="spinner-dot"></div>
    </div>
  </div>
);
