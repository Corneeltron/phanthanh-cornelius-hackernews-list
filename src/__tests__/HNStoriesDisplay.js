import renderer from 'react-test-renderer';
import { HNStoriesDisplay } from '../components/HNStoriesDisplay'

test('renders HNStoriesDisplay', () => {
  const mockIdArray = [ 34524749, 34521149, 34524006, 34523745, 34522311, 34524150 ];
  const mockCurrentPage = 2;
  const HNStoriesDisplayTree = renderer
  .create(<HNStoriesDisplay topStoriesIds={mockIdArray} currentPage={mockCurrentPage} />)
  .toJSON();
  expect(HNStoriesDisplayTree).toMatchSnapshot();
})