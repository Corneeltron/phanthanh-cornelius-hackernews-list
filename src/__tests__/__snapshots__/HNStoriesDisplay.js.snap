// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HNStoriesDisplay 1`] = `
<ol>
  <li>
    <div>
      <a
        className="links"
        href="www.ycombinator.com"
      >
        title
      </a>
      <div
        className="link-info-container"
      >
        <span
          className="link-info"
        >
          0
           points by 
        </span>
        <a
          className="link-info"
          href="https://news.ycombinator.com/user?id=default"
          rel="noreferrer"
          target="_blank"
        >
          default 
        </a>
        |
        <a
          className="link-info"
          href="https://news.ycombinator.com/item?id=0"
          rel="noreferrer"
          target="_blank"
        >
           
          0 comments
        </a>
      </div>
    </div>
  </li>
  <li>
    <div>
      <a
        className="links"
        href="www.ycombinator.com"
      >
        title
      </a>
      <div
        className="link-info-container"
      >
        <span
          className="link-info"
        >
          0
           points by 
        </span>
        <a
          className="link-info"
          href="https://news.ycombinator.com/user?id=default"
          rel="noreferrer"
          target="_blank"
        >
          default 
        </a>
        |
        <a
          className="link-info"
          href="https://news.ycombinator.com/item?id=0"
          rel="noreferrer"
          target="_blank"
        >
           
          0 comments
        </a>
      </div>
    </div>
  </li>
  <li>
    <div>
      <a
        className="links"
        href="www.ycombinator.com"
      >
        title
      </a>
      <div
        className="link-info-container"
      >
        <span
          className="link-info"
        >
          0
           points by 
        </span>
        <a
          className="link-info"
          href="https://news.ycombinator.com/user?id=default"
          rel="noreferrer"
          target="_blank"
        >
          default 
        </a>
        |
        <a
          className="link-info"
          href="https://news.ycombinator.com/item?id=0"
          rel="noreferrer"
          target="_blank"
        >
           
          0 comments
        </a>
      </div>
    </div>
  </li>
  <li>
    <div>
      <a
        className="links"
        href="www.ycombinator.com"
      >
        title
      </a>
      <div
        className="link-info-container"
      >
        <span
          className="link-info"
        >
          0
           points by 
        </span>
        <a
          className="link-info"
          href="https://news.ycombinator.com/user?id=default"
          rel="noreferrer"
          target="_blank"
        >
          default 
        </a>
        |
        <a
          className="link-info"
          href="https://news.ycombinator.com/item?id=0"
          rel="noreferrer"
          target="_blank"
        >
           
          0 comments
        </a>
      </div>
    </div>
  </li>
  <li>
    <div>
      <a
        className="links"
        href="www.ycombinator.com"
      >
        title
      </a>
      <div
        className="link-info-container"
      >
        <span
          className="link-info"
        >
          0
           points by 
        </span>
        <a
          className="link-info"
          href="https://news.ycombinator.com/user?id=default"
          rel="noreferrer"
          target="_blank"
        >
          default 
        </a>
        |
        <a
          className="link-info"
          href="https://news.ycombinator.com/item?id=0"
          rel="noreferrer"
          target="_blank"
        >
           
          0 comments
        </a>
      </div>
    </div>
  </li>
  <li>
    <div>
      <a
        className="links"
        href="www.ycombinator.com"
      >
        title
      </a>
      <div
        className="link-info-container"
      >
        <span
          className="link-info"
        >
          0
           points by 
        </span>
        <a
          className="link-info"
          href="https://news.ycombinator.com/user?id=default"
          rel="noreferrer"
          target="_blank"
        >
          default 
        </a>
        |
        <a
          className="link-info"
          href="https://news.ycombinator.com/item?id=0"
          rel="noreferrer"
          target="_blank"
        >
           
          0 comments
        </a>
      </div>
    </div>
  </li>
</ol>
`;
